Extract `ConfigReaderException` pretty printing logic to `ConfigReaderFailures`

JD557 opened the pull request:
Extracts the bulk of the `ConfigReaderException` pretty printing logic to a `prettyPrint` method on `ConfigReaderFailures` (with configurable spacing).

Closes #385

JD557 committed:
Extract `ConfigReaderException` pretty printing logic

leifwickland commented on core/src/main/scala/pureconfig/error/ConfigReaderFailures.scala:
```diff
@@ -15,6 +17,37 @@ case class ConfigReaderFailures(head: ConfigReaderFailure, tail: List[ConfigRead
   def ++(that: ConfigReaderFailures): ConfigReaderFailures =
     new ConfigReaderFailures(head, tail ++ that.toList)
+
+  def prettyPrint(identLevel: Int = 0, tabSize: Int = 2): String = {
```
Nitpick: I'd prefer to rename `tabSize` to `indentSize`.

JD557 replied:
Renamed in 5f420d0

leifwickland reviewed (approved):
@JD557 Thanks for doing this. It's a change I've wanted for a while.

Did you check if it'd make sense to add any tests now that `prettyPrint` is accessible?

JD557 committed:
Rename tabSize to identSize

JD557 committed:
Add a ConfigReaderFailuresPrettuPrintSuit

JD557 commented:
>Did you check if it'd make sense to add any tests now that prettyPrint is accessible?

I think that most pretty printing logic is already tested by the `ConfigReaderExceptionSuite`.

Nevertheless, I added some basic tests in 023f219 to a new `ConfigReaderFailuresPrettyPrintSuite` that simply checks if the identation is correctly applied.

leifwickland commented:
Thanks for adding the test @JD557. We'll wait for another maintainer to approve before merging.

ruippeixotog reviewed (commented):
@JD557 looks good, thanks for doing this!

> I think that most pretty printing logic is already tested by the ConfigReaderExceptionSuite

Nit: because we've moved this logic to another class, I'd prefer that you also moved the pretty printing logic tests to `ConfigReaderFailuresPrettyPrintSuite`, as we're not really testing the exception anymore. What do you think?

jcazevedo reviewed (approved):
@JD557: Looks good! Thanks for this!

I agree with @ruippeixotog about moving the tests related to the pretty printing logic from `ConfigReaderExceptionSuite` to `ConfigReaderFailuresPrettyPrintSuite`.

In the scope of #385, it'd also be nice to have access to the string representation (with location) of individual `ConfigReaderFailure`s, but that can be left for future work.

JD557 committed:
Move failure pretty printing related tests

JD557 commented:
>Nit: because we've moved this logic to another class, I'd prefer that you also moved the pretty printing logic tests to ConfigReaderFailuresPrettyPrintSuite, as we're not really testing the exception anymore. What do you think?

Done in 6aef1c9 (I think I moved everything).

I left the first pretty print test `ConfigReaderExceptionSuite` on purpose, to check that the exception message is actually using the pretty printer.

ruippeixotog committed:
Merge branch 'master' into move-error-messages

ruippeixotog reviewed (approved):
Excellent, thanks again!
